tensorflow/core/kernels/bincount_op.cc

```diff
@@ -75,34 +67,86 @@ class BincountOp : public OpKernel {
-    output_t->flat<T>().device(ctx->eigen_cpu_device()) =
-        partial_bins.sum(reduce_dims);
+    output.device(context->eigen_cpu_device()) = partial_bins.sum(reduce_dims);
+    return Status::OK();
   }
 };
 
-#define REGISTER(TYPE)                                               \
+}  // namespace functor
+
+template <typename Device, typename T>
+class BincountOp : public OpKernel {
+ public:
+  explicit BincountOp(OpKernelConstruction* ctx) : OpKernel(ctx) {}
+
+  void Compute(OpKernelContext* ctx) override {
+    const Tensor& arr_t = ctx->input(0);
+    const Tensor& size_tensor = ctx->input(1);
+    const Tensor& weights_t = ctx->input(2);
+
+    int32 size = size_tensor.scalar<int32>()();
+    OP_REQUIRES(ctx, size >= 0, errors::InvalidArgument(
+                                    "size (", size, ") must be non-negative"));
+
+    const bool has_weights = weights_t.NumElements() > 0;
+    OP_REQUIRES(ctx, !(has_weights && arr_t.shape() != weights_t.shape()),
+                errors::InvalidArgument(
+                    "If weights are passed, they must have the same shape (" +
+                    weights_t.shape().DebugString() + ") as arr (" +
+                    arr_t.shape().DebugString() + ")"));
+    const auto arr = arr_t.flat<int32>();
+    const auto weights = weights_t.flat<T>();
+
+    if (weights.size() == 0) {
+      Tensor* output_t;
+      OP_REQUIRES_OK(ctx,
+                     ctx->allocate_output(0, TensorShape({size}), &output_t));
+      auto output = output_t->flat<T>();
+      OP_REQUIRES_OK(
+          ctx, functor::BincountFunctor<Device, T>::Compute(ctx, arr, output));
+    } else {
+      TensorShape output_shape;
+      output_shape.AddDim(size);
+
+      Tensor* output = nullptr;
+      OP_REQUIRES_OK(ctx, ctx->allocate_output(0, output_shape, &output));
+      auto output_flat = output->flat_outer_dims<T>();
+
+      functor::UnsortedSegmentSumFunctor<Device, T, int32>()(
+          ctx, ctx->template eigen_device<Device>(), size, arr_t.shape(), arr,
+          weights.size(), weights.data(), output_flat);
+    }
+  }
+};
+
+#define REGISTER_KERNELS(type)                                       \
   REGISTER_KERNEL_BUILDER(                                           \
-      Name("Bincount").Device(DEVICE_CPU).TypeConstraint<TYPE>("T"), \
-      BincountOp<TYPE>)
+      Name("Bincount").Device(DEVICE_CPU).TypeConstraint<type>("T"), \
+      BincountOp<CPUDevice, type>)
+
+TF_CALL_NUMBER_TYPES(REGISTER_KERNELS);
+#undef REGISTER_KERNELS
+
+#if GOOGLE_CUDA
+
+#define REGISTER_KERNELS(type)                            \
+  REGISTER_KERNEL_BUILDER(Name("Bincount")                \
+                              .Device(DEVICE_GPU)         \
+                              .HostMemory("size")         \
+                              .TypeConstraint<type>("T"), \
+                          BincountOp<GPUDevice, type>)
 
-TF_CALL_NUMBER_TYPES(REGISTER);
+TF_CALL_float(REGISTER_KERNELS);
```

Comment:
Thanks. Done. Though on my dev machine it seems only int32 and float works. Might be because `sm = 20`? I added int32 to the list. Don't know if additional registration works with a different dev machine or not.